ledger/participant-integration-api/src/main/scala/platform/apiserver/TruncatedStatusInterceptor.scala

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver
+
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall
+import io.grpc.{Metadata, ServerCall, ServerCallHandler, ServerInterceptor, Status}
+
+class TruncatedStatusInterceptor(maximumDescriptionLength: Int) extends ServerInterceptor {
+  override def interceptCall[ReqT, RespT](
+      call: ServerCall[ReqT, RespT],
+      headers: Metadata,
+      next: ServerCallHandler[ReqT, RespT],
+  ): ServerCall.Listener[ReqT] =
+    next.startCall(
+      new SimpleForwardingServerCall[ReqT, RespT](call) {
+        override def close(status: Status, trailers: Metadata): Unit = {
+          val truncatedStatus = status.withDescription(truncate(status.getDescription))
+          super.close(truncatedStatus, trailers)
+        }
+      },
+      headers,
+    )
+
+  private def truncate(description: String): String =
+    Option(description)
+      .map(
+        s =>
+          if (s.length > maximumDescriptionLength)
+            s.substring(0, maximumDescriptionLength - 3) + "..."
+          else
+          s)
+      .orNull
```

Comment:
```suggestion
      if (s != null && s.length > maximumDescriptionLength)
        s.substring(0, maximumDescriptionLength - 3) + "..."
      else
        s
```
Seems simpler, feel free to drop if we have some linting check that prevents us from doing it.